DesktopModules/Vanjaro/UXManager/Extensions/Menu/GoogleReCaptcha/Controllers/SettingController.cs

```diff
@@ -10,76 +10,55 @@
 
+        [AuthorizeAccessRoles(AccessRoles = "admin")]
         [HttpPost]
-        public bool Save(dynamic Data)
+        public void Save(dynamic Data)
         {
             if (bool.Parse(Data.ApplyTo.ToString()))
             {
-                if (!string.IsNullOrEmpty(Data.Host_SiteKey.ToString()) && !string.IsNullOrEmpty(Data.Host_SecretKey.ToString()))
-                {
-                    HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.GoogleReCaptcha.SiteKey", Data.Host_SiteKey.ToString(), Config.GetDecryptionkey());
-                    HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.GoogleReCaptcha.SecretKey", Data.Host_SecretKey.ToString(), Config.GetDecryptionkey());
-                }
-                else
-                    return false;
-            }
-            else
-            {
-                if (!string.IsNullOrEmpty(Data.Site_SiteKey.ToString()) && !string.IsNullOrEmpty(Data.Site_SecretKey.ToString()))
-                {
-                    PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.GoogleReCaptcha.SiteKey", Data.Site_SiteKey.ToString(), Config.GetDecryptionkey());
-                    PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.GoogleReCaptcha.SecretKey", Data.Site_SecretKey.ToString(), Config.GetDecryptionkey());
-                }
-                else
-                    return false;
-            }
-            return true;
-        }
+                HostController hostController = new HostController();
+                hostController.UpdateEncryptedString(Core.Services.Captcha.SiteKey, Data.Host_SiteKey.ToString(), Config.GetDecryptionkey());
+                hostController.UpdateEncryptedString(Core.Services.Captcha.SecretKey, Data.Host_SecretKey.ToString(), Config.GetDecryptionkey());
+                hostController.Update(Core.Services.Captcha.Enabled, Data.Host_Enabled.ToString(), false);
 
-        [HttpPost]
-        public string Delete(dynamic Data)
-        {
-            if (bool.Parse(Data.ToString()))
-            {
-                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.GoogleReCaptcha.SiteKey", string.Empty, Config.GetDecryptionkey());
-                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.GoogleReCaptcha.SecretKey", string.Empty, Config.GetDecryptionkey());
             }
             else
             {
-                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.GoogleReCaptcha.SiteKey", string.Empty, Config.GetDecryptionkey());
-                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.GoogleReCaptcha.SecretKey", string.Empty, Config.GetDecryptionkey());
+                PortalController.UpdateEncryptedString(PortalSettings.PortalId, Core.Services.Captcha.SiteKey, Data.Site_SiteKey.ToString(), Config.GetDecryptionkey());
```

Comment:
This should also be refactored to use SettingManager. UpdatePortalSetting()

Update all code to only use SettingManager.GetPortalSetting() instead of Core.Services.Managers.SettingManager. Using Static Using.

Reply:
changes done